jpa/subsystem/src/main/java/org/jboss/as/jpa/transaction/TransactionUtil.java

```diff
@@ -45,19 +53,29 @@
 
     private static volatile TransactionSynchronizationRegistry transactionSynchronizationRegistry;
     private static volatile TransactionManager transactionManager;
+    private static volatile TransactionListenerRegistry transactionListenerRegistry;
```

Comment:
The deployed persistence unit is represented by a (PersistenceUnitServiceImpl) service but I didn't create a service for each container managed persistence context.  I suppose that could also be done with the extended persistence context as well.  The down side of using services for the container managed persistence contexts is that the number of them could be high, which at the least, could consume more constants for representing the service name (and whatever else overhead comes with those changes). Still, its an interesting idea, thanks!